packages/store/src/patterns/patternMatchers.js

```diff
@@ -1630,13 +1738,49 @@ const makePatternKit = () => {
       makeLimitsMatcher('match:bagOf', [keyPatt, countPatt, limits]),
     mapOf: (keyPatt = M.any(), valuePatt = M.any(), limits = undefined) =>
       makeLimitsMatcher('match:mapOf', [keyPatt, valuePatt, limits]),
-    split: (base, rest = undefined) =>
-      makeMatcher('match:split', rest === undefined ? [base] : [base, rest]),
-    partial: (base, rest = undefined) =>
-      makeMatcher('match:partial', rest === undefined ? [base] : [base, rest]),
+    splitArray: (base, optional = undefined, rest = undefined) =>
+      makeMatcher(
+        'match:splitArray',
+        // eslint-disable-next-line no-nested-ternary
+        optional === undefined
+          ? rest === undefined
+            ? [base]
+            : [base, [], rest]
+          : rest === undefined
+          ? [base, optional]
+          : [base, optional, rest],
+      ),
+    splitRecord: (base, optional = undefined, rest = undefined) =>
+      makeMatcher(
+        'match:splitRecord',
+        // eslint-disable-next-line no-nested-ternary
```

Comment:
consider a helper function with conditional returns instead of this nested ternary

Reply:
Done